### PR TITLE
Replace username with email

### PR DESCRIPTION
**Pull request by eduardoboucas**
This PR replaces `username` with `email` on user data. It should be merged with [this PR](https://github.com/dadiplus/publish-api/pull/6) on the API repository, which updates the password hook.

### CODE REVIEW

**Comment by eduardoboucas** [comment]
@mingard I think I might as well include in this PR the changes to move the sign in to an action, so maybe wait for that before merging?

**Comment by mingard** [comment]
@eduardoboucas cool. And we need to remove the requirement for username from API

File: app/lib/controllers/session.js [hunk 1]
```diff
@@ -5,14 +5,14 @@ const Api = require(`${paths.lib.models}/api`)
 
 const Session = function () {}
 
-Session.prototype.authorise = function (username, password, next) {
```
Comment (mingard) [inline app/lib/controllers/session.js]: @eduardoboucas has this been tested? I seem to recall an issue with changing this parameters name, and that restify wouldn't allow is as an auth middleware routine

Comment (eduardoboucas) [inline app/lib/controllers/session.js]: It has been tested. I left the argument of the `authorise` function as `username` precisely for that reason. Took me a while to figure out what was breaking though!

Comment (mingard) [inline app/lib/controllers/session.js]: Ah that was where it was. We should document that

Comment (eduardoboucas) [inline app/lib/controllers/session.js]: I think I'll add a comment in the code for now, just to make that clear?

Comment (mingard) [inline app/lib/controllers/session.js]: Yes please 🥇

**Comment by eduardoboucas** [comment]
@mingard This is ready for review.

**Pull request merged**